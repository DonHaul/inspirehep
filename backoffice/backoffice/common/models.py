from django.db import models
import uuid
from backoffice.common.constants import (
    StatusChoices,
    DEFAULT_STATUS_CHOICE,
    WorkflowType,
    DEFAULT_WORKFLOW_TYPE,
)


class AbstractWorkflow(models.Model):
    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
    data = models.JSONField()

    status = models.CharField(
        max_length=30,
        choices=StatusChoices.choices,
        default=DEFAULT_STATUS_CHOICE,
    )

    workflow_type = models.CharField(
        max_length=30,
        choices=WorkflowType.choices,
        default=DEFAULT_WORKFLOW_TYPE,
    )

    _created_at = models.DateTimeField(auto_now_add=True)
    _updated_at = models.DateTimeField(auto_now=True)

    class Meta:
        abstract = True
