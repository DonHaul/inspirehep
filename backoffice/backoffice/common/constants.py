from django.db import models


class StatusChoices(models.TextChoices):
    RUNNING = "running", "Running"
    APPROVAL = "approval", "Waiting for approval"
    PROCESSING = "processing", "Processing"
    COMPLETED = "completed", "Completed"
    ERROR = "error", "Error"


DEFAULT_STATUS_CHOICE = StatusChoices.PROCESSING


class WorkflowType(models.TextChoices):
    HEP_CREATE = "HEP_CREATE", "HEP create"
    HEP_UPDATE = "HEP_UPDATE", "HEP update"
    AUTHOR_CREATE = "AUTHOR_CREATE", "Author create"
    AUTHOR_UPDATE = "AUTHOR_UPDATE", "Author update"
    DATA_CREATE = "DATA_CREATE", "Data create"
    DATA_UPDATE = "DATA_UPDATE", "Data update"


DEFAULT_WORKFLOW_TYPE = WorkflowType.HEP_CREATE
