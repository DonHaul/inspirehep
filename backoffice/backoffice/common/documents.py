from django_opensearch_dsl import Document, fields

from backoffice.common.models import AbstractWorkflow


class WorkflowDocument(Document):
    id = fields.TextField()
    workflow_type = fields.KeywordField()
    data = fields.ObjectField(dynamic=True)
    status = fields.KeywordField()

    class Index:
        settings = {
            "number_of_shards": 1,
            "number_of_replicas": 1,
            "max_result_window": 70000,
        }

    class Django:
        model = AbstractWorkflow
        fields = [
            "_created_at",
            "_updated_at",
        ]
