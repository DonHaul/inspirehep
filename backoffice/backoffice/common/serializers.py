from rest_framework import serializers
from inspire_schemas.utils import get_validation_errors
from backoffice.common import utils
from backoffice.common.models import AbstractWorkflow


class AbstractWorkflowSerializer(serializers.ModelSerializer):
    validation_errors = serializers.JSONField(required=False)
    data = serializers.JSONField(required=True)

    def validate_data(self, value, schema):
        validation_errors = list(get_validation_errors(value, schema=schema))
        if validation_errors:
            validation_errors_msg = utils.render_validation_error_response(
                validation_errors
            )
            raise serializers.ValidationError(validation_errors_msg)
        return value

    class Meta:
        model = AbstractWorkflow
        fields = "__all__"
