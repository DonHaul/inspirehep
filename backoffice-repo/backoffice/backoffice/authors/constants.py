from django.db import models

from backoffice.workflows.workflow_types import WorkflowType

author_workflow_types = [
    WorkflowType.AUTHOR_CREATE,
    WorkflowType.AUTHOR_UPDATE,
]


class AuthorResolutionDags(models.TextChoices):
    accept = "accept", "author_create_approved_dag"
    reject = "reject", "author_create_rejected_dag"
    accept_curate = "accept_curate", "author_create_approved_dag"


class AuthorCreateDags(models.TextChoices):
    initialize = "author_create_initialization_dag", "initialize"
    approve = "author_create_approved_dag", "approve"
    reject = "author_create_rejected_dag", "reject"


class AuthorUpdateDags(models.TextChoices):
    initialize = "author_update_dag", "initialize"
