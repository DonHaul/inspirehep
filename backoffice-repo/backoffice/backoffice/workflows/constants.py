from django.db import models

from backoffice.authors.constants import (
    AuthorCreateDags,
    AuthorResolutionDags,
    AuthorUpdateDags,
)
from backoffice.workflows.workflow_types import WorkflowType

# tickets
TICKET_TYPES = (
    ("author_create_curation", "Author create curation"),
    ("author_create_user", "Author create user"),
    ("author_update_curation", "Author update curation"),
)
DEFAULT_TICKET_TYPE = "author_create_curation"

# workflows


class StatusChoices(models.TextChoices):
    RUNNING = "running", "Running"
    APPROVAL = "approval", "Waiting for approva"
    COMPLETED = "completed", "Completed"
    ERROR = "error", "Error"


DEFAULT_STATUS_CHOICE = StatusChoices.RUNNING


DEFAULT_WORKFLOW_TYPE = WorkflowType.HEP_CREATE


DECISION_CHOICES = AuthorResolutionDags.choices

WORKFLOW_DAGS = {
    WorkflowType.HEP_CREATE: "",
    WorkflowType.HEP_UPDATE: "",
    WorkflowType.AUTHOR_CREATE: AuthorCreateDags,
    WorkflowType.AUTHOR_UPDATE: AuthorUpdateDags,
}
