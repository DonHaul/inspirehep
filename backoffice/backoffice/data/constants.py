from django.db import models
from backoffice.authors.constants import WorkflowType

# tickets
TICKET_TYPES = (
    ("author_create_curation", "Author create curation"),
    ("author_create_user", "Author create user"),
    ("author_update_curation", "Author update curation"),
)
DEFAULT_TICKET_TYPE = "author_create_curation"

# workflows
allowed_workflow_types = [
    WorkflowType.AUTHOR_CREATE,
    WorkflowType.AUTHOR_UPDATE,
]


class AuthorResolutionDags(models.TextChoices):
    accept = "accept", "author_create_approved_dag"
    reject = "reject", "author_create_rejected_dag"
    accept_curate = "accept_curate", "author_create_approved_dag"


DECISION_CHOICES = AuthorResolutionDags.choices


class AuthorCreateDags(models.TextChoices):
    initialize = "author_create_initialization_dag", "initialize"
    approve = "author_create_approved_dag", "approve"
    reject = "author_create_rejected_dag", "reject"


class AuthorUpdateDags(models.TextChoices):
    initialize = "author_update_dag", "initialize"


WORKFLOW_DAGS = {
    WorkflowType.HEP_CREATE: "",
    WorkflowType.HEP_UPDATE: "",
    WorkflowType.AUTHOR_CREATE: AuthorCreateDags,
    WorkflowType.AUTHOR_UPDATE: AuthorUpdateDags,
}
