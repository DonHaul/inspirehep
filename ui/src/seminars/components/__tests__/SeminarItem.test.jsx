import React from 'react';
import { fromJS } from 'immutable';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import SeminarItem from '../SeminarItem';
import * as constants from '../../../common/constants';
import { getStore } from '../../../fixtures/store';

describe('SeminarItem', () => {
  constants.LOCAL_TIMEZONE = 'Europe/Zurich';

  it('renders with all props set', () => {
    const metadata = fromJS({
      title: { title: 'test' },
      control_number: 12345,
      can_edit: true,
      urls: [{ value: 'http://url.com' }],
      join_urls: [{ value: 'http://urljoin.com', description: 'zoom' }],
      speakers: [{ name: 'John, Doe', affiliations: [{ value: 'CERN' }] }],
      start_datetime: '2020-05-15T11:45:00.000000',
      end_datetime: '2020-05-17T00:45:00.000000',
      material_urls: [
        { value: 'http://urlmaterial.com', description: 'slides' },
      ],
    });

    const { asFragment } = render(
      <Provider store={getStore()}>
        <MemoryRouter>
          <SeminarItem metadata={metadata} />
        </MemoryRouter>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with only needed props', () => {
    const metadata = fromJS({
      title: { title: 'test' },
      control_number: 12345,
      can_edit: true,
      speakers: [{ name: 'John, Doe', affiliations: [{ value: 'CERN' }] }],
      start_datetime: '2020-05-15T11:45:00.000000',
      end_datetime: '2020-05-17T00:45:00.000000',
    });

    const { asFragment } = render(
      <Provider store={getStore()}>
        <MemoryRouter>
          <SeminarItem metadata={metadata} />
        </MemoryRouter>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with selected timezone with a different time than local timezone', () => {
    const metadata = fromJS({
      title: { title: 'test' },
      control_number: 12345,
      can_edit: true,
      speakers: [{ name: 'John, Doe', affiliations: [{ value: 'CERN' }] }],
      start_datetime: '2020-05-15T11:45:00.000000',
      end_datetime: '2020-05-17T00:45:00.000000',
    });

    const { asFragment } = render(
      <Provider store={getStore()}>
        <MemoryRouter>
          <SeminarItem metadata={metadata} selectedTimezone="America/Chicago" />
        </MemoryRouter>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with selected timezone with a same time as local timezone', () => {
    const metadata = fromJS({
      title: { title: 'test' },
      control_number: 12345,
      can_edit: true,
      speakers: [{ name: 'John, Doe', affiliations: [{ value: 'CERN' }] }],
      start_datetime: '2020-05-15T11:45:00.000000',
      end_datetime: '2020-05-17T00:45:00.000000',
    });

    const { asFragment } = render(
      <Provider store={getStore()}>
        <MemoryRouter>
          <SeminarItem metadata={metadata} selectedTimezone="Europe/Zurich" />
        </MemoryRouter>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
