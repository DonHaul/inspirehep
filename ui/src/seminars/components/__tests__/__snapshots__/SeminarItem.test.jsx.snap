// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeminarItem renders with all props set 1`] = `
<DocumentFragment>
  <div
    class="__ContentBox__ h-100"
  >
    <div
      class="ant-card ant-card-bordered h-100"
    >
      <div
        class="ant-card-body"
      >
        <div
          class="pa2"
        >
          <div>
            <div
              class="ant-row"
              type="flex"
            >
              <div
                class="ant-col"
              >
                <a
                  class="result-item-title"
                  href="/seminars/12345"
                >
                  <span
                    class="detail-page-title"
                  >
                    <span
                      class="__Latex__"
                    >
                      test
                    </span>
                  </span>
                </a>
              </div>
            </div>
            <div
              class="ant-row"
            >
              <div
                class="ant-col"
              >
                <div
                  class="__InlineList__"
                >
                  <ul>
                    <li>
                      <div
                        class="di"
                      >
                        <span>
                          John, Doe
                        </span>
                        <span
                          class="pl1 secondary-color"
                        >
                          (
                          <div
                            class="__InlineList__ di"
                          >
                            <ul>
                              <li>
                                <span>
                                  CERN
                                </span>
                              </li>
                            </ul>
                          </div>
                          )
                        </span>
                      </div>
                    </li>
                  </ul>
                </div>
              </div>
            </div>
            <div
              class="ant-row"
            >
              <div
                class="ant-col"
              >
                <span>
                  15 May 2020, 01:45 PM - 17 May 2020, 02:45 AM
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-row ant-row-space-between actions ph2"
        >
          <div
            class="ant-col"
          >
            <span
              class="__UserAction__"
            >
              <a
                href="http://url.com"
                target="_blank"
              >
                <span
                  class="__IconText__"
                >
                  <span
                    class="icon"
                  >
                    <span
                      aria-label="link"
                      class="anticon anticon-link"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="link"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M574 665.4a8.03 8.03 0 00-11.3 0L446.5 781.6c-53.8 53.8-144.6 59.5-204 0-59.5-59.5-53.8-150.2 0-204l116.2-116.2c3.1-3.1 3.1-8.2 0-11.3l-39.8-39.8a8.03 8.03 0 00-11.3 0L191.4 526.5c-84.6 84.6-84.6 221.5 0 306s221.5 84.6 306 0l116.2-116.2c3.1-3.1 3.1-8.2 0-11.3L574 665.4zm258.6-474c-84.6-84.6-221.5-84.6-306 0L410.3 307.6a8.03 8.03 0 000 11.3l39.7 39.7c3.1 3.1 8.2 3.1 11.3 0l116.2-116.2c53.8-53.8 144.6-59.5 204 0 59.5 59.5 53.8 150.2 0 204L665.3 562.6a8.03 8.03 0 000 11.3l39.8 39.8c3.1 3.1 8.2 3.1 11.3 0l116.2-116.2c84.5-84.6 84.5-221.5 0-306.1zM610.1 372.3a8.03 8.03 0 00-11.3 0L372.3 598.7a8.03 8.03 0 000 11.3l39.6 39.6c3.1 3.1 8.2 3.1 11.3 0l226.4-226.4c3.1-3.1 3.1-8.2 0-11.3l-39.5-39.6z"
                        />
                      </svg>
                    </span>
                  </span>
                  <span
                    class="v-top"
                  >
                    website
                  </span>
                </span>
              </a>
            </span>
            <span
              class="__UserAction__"
            >
              <a
                href="http://urljoin.com"
                target="_blank"
              >
                <span
                  class="__IconText__"
                >
                  <span
                    class="icon"
                  >
                    <span
                      aria-label="video-camera-add"
                      class="anticon anticon-video-camera-add"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="video-camera-add"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <defs>
                          <style />
                        </defs>
                        <path
                          d="M368 724H252V608c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8v116H72c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h116v116c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V788h116c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8z"
                        />
                        <path
                          d="M912 302.3L784 376V224c0-35.3-28.7-64-64-64H128c-35.3 0-64 28.7-64 64v352h72V232h576v560H448v72h272c35.3 0 64-28.7 64-64V648l128 73.7c21.3 12.3 48-3.1 48-27.6V330c0-24.6-26.7-40-48-27.7zM888 625l-104-59.8V458.9L888 399v226z"
                        />
                        <path
                          d="M320 360c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8H208c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h112z"
                        />
                      </svg>
                    </span>
                  </span>
                  <span
                    class="v-top"
                  >
                    join
                  </span>
                </span>
              </a>
            </span>
            <span
              class="__UserAction__"
            >
              <a
                href="http://urlmaterial.com"
                target="_blank"
              >
                <span
                  class="__IconText__"
                >
                  <span
                    class="icon"
                  >
                    <span
                      aria-label="file"
                      class="anticon anticon-file"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="file"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M854.6 288.6L639.4 73.4c-6-6-14.1-9.4-22.6-9.4H192c-17.7 0-32 14.3-32 32v832c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V311.3c0-8.5-3.4-16.7-9.4-22.7zM790.2 326H602V137.8L790.2 326zm1.8 562H232V136h302v216a42 42 0 0042 42h216v494z"
                        />
                      </svg>
                    </span>
                  </span>
                  <span
                    class="v-top"
                  >
                    material
                  </span>
                </span>
              </a>
            </span>
            <span
              class="__UserAction__"
            >
              <button
                class="ant-btn ant-btn-default ant-dropdown-trigger"
                type="button"
              >
                <span
                  class="__IconText__"
                >
                  <span
                    class="icon"
                  >
                    <span
                      aria-label="calendar"
                      class="anticon anticon-calendar"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="calendar"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
                        />
                      </svg>
                    </span>
                  </span>
                  <span
                    class="v-top"
                  >
                    export
                  </span>
                </span>
              </button>
            </span>
            <span
              class="__UserAction__"
            >
              <a
                href="/submissions/seminars/12345"
                target="_blank"
              >
                <span
                  class="__IconText__"
                >
                  <span
                    class="icon"
                  >
                    <span
                      aria-label="edit"
                      class="anticon anticon-edit"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="edit"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 000-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 009.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
                        />
                      </svg>
                    </span>
                  </span>
                  <span
                    class="v-top"
                  >
                    edit
                  </span>
                </span>
              </a>
            </span>
          </div>
          <div
            class="ant-col"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`SeminarItem renders with only needed props 1`] = `
<DocumentFragment>
  <div
    class="__ContentBox__ h-100"
  >
    <div
      class="ant-card ant-card-bordered h-100"
    >
      <div
        class="ant-card-body"
      >
        <div
          class="pa2"
        >
          <div>
            <div
              class="ant-row"
              type="flex"
            >
              <div
                class="ant-col"
              >
                <a
                  class="result-item-title"
                  href="/seminars/12345"
                >
                  <span
                    class="detail-page-title"
                  >
                    <span
                      class="__Latex__"
                    >
                      test
                    </span>
                  </span>
                </a>
              </div>
            </div>
            <div
              class="ant-row"
            >
              <div
                class="ant-col"
              >
                <div
                  class="__InlineList__"
                >
                  <ul>
                    <li>
                      <div
                        class="di"
                      >
                        <span>
                          John, Doe
                        </span>
                        <span
                          class="pl1 secondary-color"
                        >
                          (
                          <div
                            class="__InlineList__ di"
                          >
                            <ul>
                              <li>
                                <span>
                                  CERN
                                </span>
                              </li>
                            </ul>
                          </div>
                          )
                        </span>
                      </div>
                    </li>
                  </ul>
                </div>
              </div>
            </div>
            <div
              class="ant-row"
            >
              <div
                class="ant-col"
              >
                <span>
                  15 May 2020, 01:45 PM - 17 May 2020, 02:45 AM
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-row ant-row-space-between actions ph2"
        >
          <div
            class="ant-col"
          >
            <span
              class="__UserAction__"
            >
              <button
                class="ant-btn ant-btn-default ant-dropdown-trigger"
                type="button"
              >
                <span
                  class="__IconText__"
                >
                  <span
                    class="icon"
                  >
                    <span
                      aria-label="calendar"
                      class="anticon anticon-calendar"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="calendar"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
                        />
                      </svg>
                    </span>
                  </span>
                  <span
                    class="v-top"
                  >
                    export
                  </span>
                </span>
              </button>
            </span>
            <span
              class="__UserAction__"
            >
              <a
                href="/submissions/seminars/12345"
                target="_blank"
              >
                <span
                  class="__IconText__"
                >
                  <span
                    class="icon"
                  >
                    <span
                      aria-label="edit"
                      class="anticon anticon-edit"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="edit"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 000-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 009.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
                        />
                      </svg>
                    </span>
                  </span>
                  <span
                    class="v-top"
                  >
                    edit
                  </span>
                </span>
              </a>
            </span>
          </div>
          <div
            class="ant-col"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`SeminarItem renders with selected timezone with a different time than local timezone 1`] = `
<DocumentFragment>
  <div
    class="__ContentBox__ h-100"
  >
    <div
      class="ant-card ant-card-bordered h-100"
    >
      <div
        class="ant-card-body"
      >
        <div
          class="pa2"
        >
          <div>
            <div
              class="ant-row"
              type="flex"
            >
              <div
                class="ant-col"
              >
                <a
                  class="result-item-title"
                  href="/seminars/12345"
                >
                  <span
                    class="detail-page-title"
                  >
                    <span
                      class="__Latex__"
                    >
                      test
                    </span>
                  </span>
                </a>
              </div>
            </div>
            <div
              class="ant-row"
            >
              <div
                class="ant-col"
              >
                <div
                  class="__InlineList__"
                >
                  <ul>
                    <li>
                      <div
                        class="di"
                      >
                        <span>
                          John, Doe
                        </span>
                        <span
                          class="pl1 secondary-color"
                        >
                          (
                          <div
                            class="__InlineList__ di"
                          >
                            <ul>
                              <li>
                                <span>
                                  CERN
                                </span>
                              </li>
                            </ul>
                          </div>
                          )
                        </span>
                      </div>
                    </li>
                  </ul>
                </div>
              </div>
            </div>
            <div
              class="ant-row"
            >
              <div
                class="ant-col"
              >
                <span
                  class="red"
                >
                  15 May 2020, 06:45 AM - 16 May 2020, 07:45 PM CST
                </span>
                 (
                <span>
                  15 May 2020, 01:45 PM - 17 May 2020, 02:45 AM CET
                </span>
                )
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-row ant-row-space-between actions ph2"
        >
          <div
            class="ant-col"
          >
            <span
              class="__UserAction__"
            >
              <button
                class="ant-btn ant-btn-default ant-dropdown-trigger"
                type="button"
              >
                <span
                  class="__IconText__"
                >
                  <span
                    class="icon"
                  >
                    <span
                      aria-label="calendar"
                      class="anticon anticon-calendar"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="calendar"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
                        />
                      </svg>
                    </span>
                  </span>
                  <span
                    class="v-top"
                  >
                    export
                  </span>
                </span>
              </button>
            </span>
            <span
              class="__UserAction__"
            >
              <a
                href="/submissions/seminars/12345"
                target="_blank"
              >
                <span
                  class="__IconText__"
                >
                  <span
                    class="icon"
                  >
                    <span
                      aria-label="edit"
                      class="anticon anticon-edit"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="edit"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 000-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 009.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
                        />
                      </svg>
                    </span>
                  </span>
                  <span
                    class="v-top"
                  >
                    edit
                  </span>
                </span>
              </a>
            </span>
          </div>
          <div
            class="ant-col"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`SeminarItem renders with selected timezone with a same time as local timezone 1`] = `
<DocumentFragment>
  <div
    class="__ContentBox__ h-100"
  >
    <div
      class="ant-card ant-card-bordered h-100"
    >
      <div
        class="ant-card-body"
      >
        <div
          class="pa2"
        >
          <div>
            <div
              class="ant-row"
              type="flex"
            >
              <div
                class="ant-col"
              >
                <a
                  class="result-item-title"
                  href="/seminars/12345"
                >
                  <span
                    class="detail-page-title"
                  >
                    <span
                      class="__Latex__"
                    >
                      test
                    </span>
                  </span>
                </a>
              </div>
            </div>
            <div
              class="ant-row"
            >
              <div
                class="ant-col"
              >
                <div
                  class="__InlineList__"
                >
                  <ul>
                    <li>
                      <div
                        class="di"
                      >
                        <span>
                          John, Doe
                        </span>
                        <span
                          class="pl1 secondary-color"
                        >
                          (
                          <div
                            class="__InlineList__ di"
                          >
                            <ul>
                              <li>
                                <span>
                                  CERN
                                </span>
                              </li>
                            </ul>
                          </div>
                          )
                        </span>
                      </div>
                    </li>
                  </ul>
                </div>
              </div>
            </div>
            <div
              class="ant-row"
            >
              <div
                class="ant-col"
              >
                <span>
                  15 May 2020, 01:45 PM - 17 May 2020, 02:45 AM
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-row ant-row-space-between actions ph2"
        >
          <div
            class="ant-col"
          >
            <span
              class="__UserAction__"
            >
              <button
                class="ant-btn ant-btn-default ant-dropdown-trigger"
                type="button"
              >
                <span
                  class="__IconText__"
                >
                  <span
                    class="icon"
                  >
                    <span
                      aria-label="calendar"
                      class="anticon anticon-calendar"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="calendar"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
                        />
                      </svg>
                    </span>
                  </span>
                  <span
                    class="v-top"
                  >
                    export
                  </span>
                </span>
              </button>
            </span>
            <span
              class="__UserAction__"
            >
              <a
                href="/submissions/seminars/12345"
                target="_blank"
              >
                <span
                  class="__IconText__"
                >
                  <span
                    class="icon"
                  >
                    <span
                      aria-label="edit"
                      class="anticon anticon-edit"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="edit"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 000-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 009.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
                        />
                      </svg>
                    </span>
                  </span>
                  <span
                    class="v-top"
                  >
                    edit
                  </span>
                </span>
              </a>
            </span>
          </div>
          <div
            class="ant-col"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
